sdk/eventhub/Azure.Messaging.EventHubs.Processor/tests/Snippets/Sample07_MockEventProcessorHandlersTests.cs

```diff
@@ -0,0 +1,244 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Core;
+using Azure.Core.Pipeline;
+using Azure.Messaging.EventHubs.Consumer;
+using Azure.Messaging.EventHubs.Processor;
+using Azure.Storage.Blobs;
+using Moq;
+using NUnit.Framework;
+
+namespace Azure.Messaging.EventHubs.Tests.Snippets
+{
+    /// <summary>
+    ///   The suite of tests defining the snippets used in the Event Hubs
+    ///   Sample07_MockEventProcessorHandlersTests sample.
+    /// </summary>
+    ///
+    [TestFixture]
+    [Category(TestCategory.Live)]
+    [Category(TestCategory.DisallowVisualStudioLiveUnitTesting)]
+    public class Sample07_MockEventProcessorHandlersTests
+    {
+        /// <summary>
+        ///   Performs basic smoke test validation of the contained snippet.
+        /// </summary>
+        ///
+        [Test]
+        public async Task ProcessEventHandlerCheckpoint()
+        {
+            #region Snippet:EventHubs_Processor_Sample07_ProcessEventHandlerCheckpoint
+
+            // Application handler checkpoints after processing defined number of events.
+
+            const int EventsBeforeCheckpoint = 2;
+            var partitionEventCount = new ConcurrentDictionary<string, int>();
+            bool checkpointUpdated = false;
+
+            async Task processEventHandler(ProcessEventArgs args)
+            {
+                try
+                {
+                    checkpointUpdated = false;
+
+                    await Application.ProcessEventAsync(args);
+
+                    string partition = args.Partition.PartitionId;
+
+                    int eventsSinceLastCheckpoint = partitionEventCount.AddOrUpdate(
+                        key: partition,
+                        addValue: 1,
+                        updateValueFactory: (_, currentCount) => currentCount + 1);
+
+                    if (eventsSinceLastCheckpoint >= EventsBeforeCheckpoint)
+                    {
+                        await args.UpdateCheckpointAsync();
+                        partitionEventCount[partition] = 0;
+                    }
+                }
+                catch (Exception ex)
+                {
+                    Application.HandleProcessingException(args, ex);
+                }
+            }
+
+            // Initialize the args and simulate checkpoint update method.
+
+            var partitionId = "0";
+            var eventBody = new BinaryData("This is a sample event body");
+            var eventData = EventHubsModelFactory.EventData(eventBody);
```

Comment:
We should consider expanding this a bit to include the properties that would be on an event received and commenting that is why we're using the model factory.   Right now, there's no difference from using the constructor directly.   

We probably don't need to fill out ALL the properties, maybe just Offset and Sequence number; the comment can stand as starting point for "look at this method to populate all those read-only fields on Event Data"